src/RecordManager/Base/Command/Records/Export.php

```diff
@@ -335,8 +368,90 @@ function ($record) use (
             );
             $returnCode = Command::FAILURE;
         }
-        file_put_contents($file, "</collection>\n", FILE_APPEND);
+        $this->finishBatch();
 
         return $returnCode;
     }
+
+    /**
+     * Start a new batch
+     *
+     * @return void
+     */
+    protected function startNewBatch(): void
+    {
+        if ($this->currentFile) {
+            $this->finishBatch();
+        }
+        $this->currentBatch += 1;
+        $this->currentBatchCount = 0;
+        $this->currentFile = $this->getBatchFileName();
+        $this->logger
+            ->logInfo('exportRecords', "Exporting to file: $this->currentFile");
+        if (file_exists($this->currentFile)) {
+            unlink($this->currentFile);
+        }
+        file_put_contents(
+            $this->currentFile,
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n\n<collection>\n",
+            FILE_APPEND
+        );
+    }
+
+    /**
+     * Finish writing to a batch
+     *
+     * @return void
+     */
+    protected function finishBatch(): void
+    {
+        if (!$this->currentFile) {
+            throw new \Exception('Batch not properly started');
+        }
+        file_put_contents($this->currentFile, "</collection>\n", FILE_APPEND);
+        $this->currentFile = null;
+    }
+
+    /**
+     * Get a batch file name
+     *
+     * @return string
+     */
+    protected function getBatchFileName(): string
+    {
+        if ('-' === $this->fileTemplate) {
+            return 'php://stdout';
+        }
+
+        if (!$this->batchSize) {
+            return $this->fileTemplate;
+        }
+        $result
+            = str_replace('{n}', (string)$this->currentBatch, $this->fileTemplate);
+        if ($result === $this->fileTemplate) {
+            $result .= "-$this->currentBatch";
+        }
+        return $result;
+    }
+
+    /**
+     * Write a record
+     *
+     * Start a new batch if necessary.
+     *
+     * @param string $record Record
+     *
+     * @return void
+     */
+    protected function writeRecord(string $record): void
+    {
+        ++$this->currentBatchCount;
```

Comment:
Maybe it would make sense to check for end of batch before incrementing currentBatchCount so you don't have to assign it to 1 after assigning it to 0 in startNewBatch... I found that a little confusing at first. :-)

Reply:
Not sure why I did that, and I wasn't happy with it even then. Now fixed, thanks! :)